### PR TITLE
kogito-serverless-operator-343: Move workflows execution JDK to java17

### PR DESCRIPTION
**Pull request by wmedvede**
**Description of the change:**

**Motivation for the change:**

**Checklist**

- [X] Add or Modify a unit test for your change
- [X] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by wmedvede** [comment]
closes: #343

File: container-builder/examples/dockerfiles/SonataFlow.dockerfile [hunk 1]
```diff
@@ -30,7 +30,7 @@ RUN /home/kogito/launch/build-app.sh ./resources
 #=============================
 # Runtime Run
 #=============================
-FROM registry.access.redhat.com/ubi8/openjdk-11-runtime:latest
+FROM registry.access.redhat.com/ubi8/openjdk-17:latest
```
Comment (jakubschwan) [inline container-builder/examples/dockerfiles/SonataFlow.dockerfile:33]: Here is change of base image type, is this change needed? Before was used runtime-only container.

**Review by jakubschwan — COMMENTED** [review]
For SonataFlow dockerfile is with update to JDK 17 also changed base image type

File: container-builder/examples/dockerfiles/SonataFlow.dockerfile (hunk 1 above)
Comment (wmedvede) [reply container-builder/examples/dockerfiles/SonataFlow.dockerfile:33]: This just an example of a potential dockerfile, so I basically unified the configurations.

Comment (wmedvede) [reply container-builder/examples/dockerfiles/SonataFlow.dockerfile]: anyway, I have configuerd openjdk-17-runtime instead

**Review by ricardozanini — APPROVED** [review]
Hmmmj should have done this as part of my work  on the images. I think we were too eager to merge there. ☹️

File: container-builder/builder/kubernetes/testdata/Dockerfile [hunk 2]
```diff
@@ -32,7 +32,7 @@ RUN ${MAVEN_HOME}/bin/mvn -U -B ${MAVEN_ARGS_APPEND} -s ${MAVEN_SETTINGS_PATH} c
 #=============================
 # Runtime Run
 #=============================
-FROM registry.access.redhat.com/ubi8/openjdk-11:1.11
+FROM registry.access.redhat.com/ubi8/openjdk-17:1.18-2
```
Comment (jakubschwan) [inline container-builder/builder/kubernetes/testdata/Dockerfile:35]: use abstract 1.18 tag instead of specific 1.18-2 , in case another 1.18 build or CVE fix, the latest will be used and image up to date. 
```suggestion
FROM registry.access.redhat.com/ubi8/openjdk-17:1.18
```

File: test/builder/Dockerfile [hunk 3]
```diff
@@ -32,7 +32,7 @@ RUN ${MAVEN_HOME}/bin/mvn -U -B ${MAVEN_ARGS_APPEND} -s ${MAVEN_SETTINGS_PATH} c
 #=============================
 # Runtime Run
 #=============================
-FROM registry.access.redhat.com/ubi8/openjdk-11:1.11
+FROM registry.access.redhat.com/ubi8/openjdk-17:1.18-2
```
Comment (jakubschwan) [inline test/builder/Dockerfile:35]:
```suggestion
FROM registry.access.redhat.com/ubi8/openjdk-17:1.18
```

**Review by jakubschwan — CHANGES_REQUESTED** [review]
Also please update the version tags, right now are used specific tags that might not be update in case of rebuild or CVE update.
Also do we want to have set specific tags for builders or use latests? (We can open a new ticket for builder update tag to latest as well)

**Review by jakubschwan — APPROVED** [review]
